### ISSUE TITLE
lnd terminates after error -26

### ISSUE THREAD

**Issue by romanoLT** [lightningnetwork/lnd#4888]
### Background

I tried to 'revive' an old node which was running LND v0.7ish and was offline for some time. I upgraded to v0.12 beta rc2. 
I got an error which seems to be critical because it shuts down my node.

### Your environment

* version of `lnd` : 0.12 beta rc2
* which operating system (`uname -a` on *Nix): Debian 10, 4.19 and 64
* version of `btcd`, `bitcoind`, or other backend: bitcoind 0.20.1
* any other relevant environment details

### Steps to reproduce

Unlock the wallet.

### Expected behaviour

Just broadcast channel closing and not terminate LND instance.

### Actual behaviour

once lncli unlock is ruined : LND displays these messages:
 [INF] CNCT: Creating ChannelArbitrators for 5 active channels
 [INF] CNCT: Re-publishing force close tx(ccccc...) for channel abc44...:0
 [INF] LNWL: Inserting unconfirmed transaction ccccc...
 [INF] LNWL: Removed invalid transaction: (*wire.MsgTx)(0xc001736200)({....

and then :
[ERR] CNCT: ChannelArbitrator(ccccc... :0): unable to broadcast close tx: unmatched backend error: -26: non-mandatory-script-verify-flag (Signature must be zero for failed CHECK(MULTI)SIG operation)

**Comment by Roasbeef** [comment]
How long was the node offline for? Do you have any other earlier/surrounding logs? Sub-systems of interest are `CNCT`, `LNWL`, and `PEER`.

**Comment by romanoLT** [comment]
@Roasbeef 
I would say more than year (summer 2019) but I'm not one hundred percent sure.
Here is enclosed an extract of the log file 
[lnd extract.log](https://github.com/lightningnetwork/lnd/files/5750027/lnd.extract.log)

**Comment by romanoLT** [comment]
Well I tried with newer version of lnd but problem remains. Is there a way to cancel the broadcast of this incorrection tx that breaks lnd?

**Comment by guggero** [comment]
Can you check whether the funding transaction was already spent? Possibly your remote peer force closed on you a while ago and that transaction wouldn't be valid anyway.
Do you have a `channel.backup` from last year? Not sure if that was already supported by 0.7 but I think so.
If you do and if the remote peer force closed on you, I'd recommend the following steps:
1. Make a copy of the `channel.backup` file outside of the `.lnd` directory (the next steps will overwrite the default one!)
2. Stop `lnd`.
3. Download [`chantools`](https://github.com/guggero/chantools/releases/tag/v0.8.2) and run [`chantools removechannel`](https://github.com/guggero/chantools/blob/master/doc/chantools_removechannel.md) on the failing channel.
4. If there was no balance on your side (which it looks like from the force close TX, it's only one output and it's P2WKH so the `to_remote` unencumbered output), you're done.
5. If there was balance on your side, please comment here again for further steps.

**Comment by romanoLT** [comment]
Hi @guggero, sorry for the late reply. So it seems I do have a channel.backup file.
I removed one channel and and was able to start but lnd is still throwing an error 26 (which doesn't kill the daemon). I have enclosed a log file.
Also, I'm missing some funds I guess they are in a local balance. What would you advise to retrieve these funds from the lightning network to my btc wallet?
Thanks

**Comment by guggero** [comment]
If you continue to get that error then you have to repeat the `chantools removechannel` step until `lnd` can start up correctly.

To get your funds back, you have three options:
1. If there are no open channels (only pending) left after you removed the non-functioning ones, you can just run `lncli restorechanbackup --multi_file <the_channel.backup_file_you_copied_earlier>`. That should bring back the balances of the channels you removed (but it will take a while as a lot of chain-rescanning is involved, just let it run for a day or so).
2. If you don't plan on continuing to run `lnd` and just want to get your funds out, you can also just follow the [recovery guide](https://docs.lightning.engineering/advanced-best-practices/advanced-best-practices/recovery) that will close out all channels.
1. If there are any channels left after you removed the non-functioning ones and you want to continue operating the node, there's a way you can do that but it involves a few more manual steps. Basically filtering the `channel.backup` file until only the ones you removed before remain and then restore just those channels. Contact me on Slack if you choose that option.

**Comment by guggero** [comment]
We looked into the issue of the invalid transaction and couldn't find out what could cause this. Neither of the CVEs apply.
Since this is from a pre-0.7.0 `lnd` node and this doesn't seem to appear often in the wild, this might just be data corruption in the channel DB.
The two affected channels with the invalid transaction only have remote balance. The one channel that has local balance was force closed successfully and I'm working with the user to sweep the funds.

If anyone of the team wants to take a look at the invalid transactions, I have the raw hex of them.
Closing this as there's currently nothing more we can do.
